### PR TITLE
Add variant support in download-frozen-image-v2.sh

### PR DESCRIPTION
**Pull request by Pennyzct**
Since for now download-frozen-image-v2.sh won't be smart enough to use the adequate image in arm platform, hence we add variant support to fix it. Relevant request has been raised in github https://github.com/moby/moby/pull/35929 https://github.com/moby/moby/issues/34875. We have also pulled similar request for adding variant-matching when pulling Multi-arch image. open https://github.com/moby/moby/pull/36121

**- What I did**
Add variant support when downloading frozen image in arm platform 
**- How I did it**
Read from /proc/cpuinfo to get variant info, then add variant-matching when downloading arm frozen-image. This only work for linux. 
**- How to verify it**
We can verify it directly with make binary.

**- A picture of a cute animal (not mandatory but encouraged**
image

### CODE REVIEW

**Comment by Pennyzct** [comment]
@arm64b @weichen81 @thaJeztah @tianon

File: contrib/download-frozen-image-v2.sh [hunk 1]
```diff
@@ -225,32 +256,57 @@ while [ $# -gt 0 ]; do
...
-									-H 'Accept: application/vnd.docker.distribution.manifest.v1+json' \
-									"$registryBase/v2/$image/manifests/$digest"
-							)"
-							handle_single_manifest_v2 "$submanifestJson"
-							found="found"
-							break
-						fi
-					done
+					if [ "${Arch}" = "arm64" ] || [ "${Arch}" = "arm" ]; then
+                                                fetch_arm_variant
```
Comment (thaJeztah) [inline contrib/download-frozen-image-v2.sh]: Looks like this line uses spaces, not tabs, which makes @tianon cry :wink:

File: contrib/download-frozen-image-v2.sh [hunk 2]
```diff
@@ -47,6 +47,37 @@ registryBase='https://registry-1.docker.io'
 authBase='https://auth.docker.io'
 authService='registry.docker.io'
 
+ARM_VARIANT=""
+
+fetch_arm_variant() {
+        if [ "$(go env GOHOSTOS)" = "linux" ];then
```
Comment (thaJeztah) [inline contrib/download-frozen-image-v2.sh]: Can you change this to use tabs for indentation?

**Review by thaJeztah — CHANGES_REQUESTED** [review]
@tianon PTAL

File: contrib/download-frozen-image-v2.sh (hunk 2 above)
Comment (Pennyzct) [reply contrib/download-frozen-image-v2.sh]: sorry~Fix it now😊

File: contrib/download-frozen-image-v2.sh (hunk 1 above)
Comment (Pennyzct) [reply contrib/download-frozen-image-v2.sh]: sorry~Fix it now😊

Comment (thaJeztah) [reply contrib/download-frozen-image-v2.sh]: 🤗

File: contrib/download-frozen-image-v2.sh [hunk 3]
```diff
@@ -47,6 +47,37 @@ registryBase='https://registry-1.docker.io'
 authBase='https://auth.docker.io'
 authService='registry.docker.io'
 
+ARM_VARIANT=""
+
+fetch_arm_variant() {
+	if [ "$(go env GOHOSTOS)" = "linux" ];then
+		local cpuinfo="$(cat /proc/cpuinfo | grep -i 'CPU architecture' | head -n 1 -)"
+		local variant="$(echo -e "${cpuinfo##*:}" | tr -d '[:space:]')"
```
Comment (arm64b) [inline contrib/download-frozen-image-v2.sh]: So this PR tries to pull the image with the OSType/Arch abstracted from the host which is running the code here, while not  to specify a different Arch than the local host.

**Review by arm64b — COMMENTED** [review]
Do we need to add the failure fallback mode if the image we want to pull has not exact arch match with the local, but it's compatible meaning can be running on current local host?

**Pull request closed without merging**